### ISSUE TITLE
[DX] Make the score_fn follow the scikit-learn standard to avoid confusions

### ISSUE THREAD

**Issue by ancestor-mithril** [fii-optim-lab/hgp-lib#93]
Current `score_fn` requires predictions first, and then labels, while the sklearn standard is `y_true`, `y_pred`. 
Users are required to "reverse" the GT with the prediction in the current version.
